util/sparse/punch_unix.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Tailscale Inc & AUTHORS
```

Comment:
Let's add a single `punch.go` file that has:
```go
// Package sparse contains functionality for interacting with sparse files.
package sparse

// PunchAt ...
func PunchAt(fd *os.File, off, size int64) error {
    return punchAt(fd, off, size)
}
```

Thus, we can avoid redundant external documentation in each of the OS-specific files.
We should have this document what guarantees we can expect. For example, if sparse files isn't supported, does it just implicitly just write zeros? or does it report something like `errors.ErrUnsupported`?

We should also add a `punch_generic.go` that is used for non-Windows, non-Linux, non-Darwin, and non-Unix that implements the default fallback.